eks_fargate/datadog_checks/eks_fargate/eks_fargate.py

```diff
@@ -1,31 +1,141 @@
 # (C) Datadog, Inc. 2020
 # All rights reserved
 # Licensed under a 3-clause BSD style license (see LICENSE)
+import json
+import logging
 import os
+from datetime import datetime, timedelta
+
+import requests
+from kubeutil import get_connection_info
 
 from datadog_checks.base import AgentCheck
+from datadog_checks.base.errors import CheckException
+from datadog_checks.base.utils.date import UTC
+from datadog_checks.base.utils.tagging import tagger
+
+from .common import ExpiredPodFilter, KubeletCredentials
+
+try:
+    from datadog_agent import get_config
+except ImportError:
+
+    def get_config(key):
+        return ""
+
+
+KUBELET_NODE_ENV_VAR = 'DD_KUBERNETES_KUBELET_NODENAME'
+HOSTNAME_ENV_VAR = 'HOSTNAME'
+CAPACITY_ANNOTATION_KEY = 'CapacityProvisioned'
+POD_LIST_PATH = '/pods'
+
+log = logging.getLogger('collector')
 
 
 class EksFargateCheck(AgentCheck):
     """
-    Generate a heartbeat for Amazon EKS on AWS Fargate workloads
+    Generate heartbeat and capacity metrics for Amazon EKS on AWS Fargate workloads
     """
 
     def __init__(self, name, init_config, instances):
         super(EksFargateCheck, self).__init__(name, init_config, instances)
-        pod_name = os.getenv("HOSTNAME")
-        virtual_node = os.getenv("DD_KUBERNETES_KUBELET_NODENAME", "")
+        self.NAMESPACE = 'eks.fargate'
+
+        pod_name = os.getenv(HOSTNAME_ENV_VAR)
+        virtual_node = os.getenv(KUBELET_NODE_ENV_VAR, "")
+        self.fargate_mode = 'fargate' in virtual_node and pod_name is not None
 
-        self.fargate_mode = 'fargate' in virtual_node
+        kubelet_conn_info = get_connection_info()
+        endpoint = kubelet_conn_info.get('url')
+        if endpoint is None:
+            raise CheckException("Unable to detect the kubelet URL automatically: " + kubelet_conn_info.get('err', ''))
 
-        if pod_name is not None and self.fargate_mode:
+        self.pod_list_url = endpoint.strip("/") + POD_LIST_PATH
+        self.kubelet_credentials = KubeletCredentials(kubelet_conn_info)
+        self.pod_list = self.get_pod_list()
+
+        if self.fargate_mode:
             self.tags = []
-            self.tags.append("pod_name:" + pod_name)
-            self.tags.append("virtual_node:" + virtual_node)
+            self.tags.append('pod_name:' + pod_name)
```

Comment:
I think we don't have to add the `pod_name` tag manually anymore, it should be added by the tagger now

Reply:
that said, we don't have to read it from the hostname env var anymore